galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/GherkinTestRunner.java

```diff
@@ -457,18 +458,18 @@ private void markWaiting(@NotNull IFramework framework) throws TestRunException
         }
     }
 
-    private void updateStatus(String status, String timestamp) throws TestRunException {
+    private void updateStatus(TestRunLifecycleStatus status, String timestamp) throws TestRunException {
 
-        this.testStructure.setStatus(status);
-        if ("finished".equals(status)) {
+        this.testStructure.setStatus(status.toString());
+        if ("finished".equals(status.toString())) {
             updateResult();
             this.testStructure.setEndTime(Instant.now());
         }
 
         writeTestStructure();
 
         try {
-            this.dss.put("run." + run.getName() + ".status", status);
+            this.dss.put("run." + run.getName() + ".status", status.toString());
             if (timestamp != null) {
                 this.dss.put("run." + run.getName() + "." + timestamp, Instant.now().toString());
```

Comment:
Another use of Instant.now() ...

Reply:
And `"run." + run.getName() + "."` is duplicated 3 times in just those 4 lines ?

Reply:
new ticket raised to deal with this